TFLu: Add stm32f4 and build target

mansnils opened the pull request:
Add new TARGET=stm32f4 that is working with Renode.
Add new <build> target that will just build the test binaries.
Add new CI script for this as well.
The purpose of this is CMSIS-NN regression.

mansnils committed:
TFLu: Add stm32f4 and build target

Add new TARGET=stm32f4 that is working with Renode.
Add new <build> target that will just build the test binaries.
Add new CI script for this as well.
The purpose of this is CMSIS-NN regression.

mansnils commented:
Gentle ping for review.

petewarden commented on tensorflow/lite/micro/tools/make/targets/stm32f4_makefile.inc:
```diff
@@ -0,0 +1,86 @@
+    $(MAKEFILE_DIR)/downloads/stm32_bare_lib/source/debug_log.c
+  MICROLITE_CC_SRCS := $(filter-out $(EXCLUDED_SRCS), $(MICROLITE_CC_SRCS))
+  # Stm32f4 is reusing the bluepill renode scripts for now
+  TEST_SCRIPT := tensorflow/lite/micro/testing/test_bluepill_binary.sh
```
This PR looks great overall, but I'm confused about how the runtime testing with Renode works. I thought the test_bluepill_binary.sh script invoked Renode with a Cortex M3 target, but it looks like you're building for a Cortex M4? Can you help me understand what's going on here? Thanks!

mansnils replied:
Well in the beginning we decided to introduce a new M4 target since we knew that was working on real HW on that particular branch/commit and we didn't know the last time when the Bluepill target was working. Then it turns out the two targets are quite similar from Renode point of view. Looking in ./tensorflow/lite/micro/testing/bluepill.resc, there is the line: machine LoadPlatformDescription @platforms/cpus/stm32f103.repl.
When I started testing, I used: machine LoadPlatformDescription @platforms/cpus/stm32f4.repl, which I would expect for M4, and that is about the only difference. However it turned out it doesn't really matter so I decided not to copy every Bluepill file, which would otherwise be required. Michael Gielda at Antmicro, answered: "Well, how those TF Lite Micro examples work, it probably does not really matter which CPU core they run on, but of course I can ask the team. Your cores are pretty compatible if you don't use the fancy stuff."

mansnils replied:
After all we decided to copy new files from the bluepill files to get the correct platform description for M4.
We may use the "fancy stuff" as there a lot of optimizations in CMSIS, and it would also otherwise be confusing for a user.

mansnils committed:
TFLu: Update stm32f4 target

Add stm32f4 specific renode script files, instead of reusing the
bluepill files.

mansnils commented:
Ready for merge

mansnils commented:
Ready for merge

rthadur commented on tensorflow/lite/micro/testing/test_stm32f4_binary.sh:
```diff
@@ -0,0 +1,60 @@
+declare -r MICRO_LOG_FILENAME=${MICRO_LOG_PATH}/logs.txt
+mkdir -p ${MICRO_LOG_PATH}
+
+docker build -t renode_stm32f4 \
```
`tensorflow/lite/micro/tools/make/Makefile:250: warning: overriding recipe for target 'tensorflow/lite/micro/tools/make/downloads/cmsis'
tensorflow/lite/micro/tools/make/Makefile:250: warning: ignoring old recipe for target 'tensorflow/lite/micro/tools/make/downloads/cmsis'
tensorflow/lite/micro/testing/test_stm32f4_binary.sh tensorflow/lite/micro/tools/make/gen/stm32f4_cortex-m4/bin/hello_world_test '~~~ALL TESTS PASSED~~~'
tensorflow/lite/micro/testing/test_stm32f4_binary.sh: line 33: docker: command not found
tensorflow/lite/micro/examples/hello_world/Makefile.inc:33: recipe for target 'test_hello_world_test' failed
make: *** [test_hello_world_test] Error 127`

mansnils replied:
I think that is related to hello_world_test binary is not building due to RAM being too small that I saw earlier. Although I did no longer see the error on the latest master branch. Increased RAM size anyway.

rthadur reviewed (changes requested):
pasted internal error , please check ?

mansnils committed:
TFLu: Update stm32f4 target

Filter out failed test and increase RAM size.

mansnils commented:
Do you know why these tests fail? They didn't fail on the first and second commit, so I wonder why they fail now.
